internal/bminventory/inventory.go

```diff
@@ -4669,6 +4701,19 @@ func (b *bareMetalInventory) BindHostInternal(ctx context.Context, params instal
 		return nil, common.NewApiError(http.StatusInternalServerError, err)
 	}
 
+	//At this point the host is updated. Retrigger auto-assign role
+	//calculation since the new host may affect the assignment.
+	if cluster != nil {
+		updated, as_err := b.clusterApi.ResetAutoAssignRoles(ctx, cluster, transaction.AddForUpdateQueryOption(b.db))
+		if as_err != nil {
+			return nil, common.NewApiError(http.StatusInternalServerError, as_err)
+		}
+		if updated {
+			log.Infof("re-triggering auto role assignment for cluster %s following host binding", cluster.ID.String())
+			_ = b.updateHostsAndClusterStatus(ctx, cluster, b.db, log)
```

Comment:
no need for err handling?

Reply:
I will log the error but I wouldn't want to fail the registration process. A subsequent refresh might fix the problem